xplique/attributions/forgrad.py

```diff
@@ -0,0 +1,90 @@
+"""
+Module related to ForGrad method enchancement
+"""
+
+from functools import lru_cache
+
+import tensorflow as tf
+import numpy as np
+
+
+@lru_cache(maxsize=32)
+def _low_pass_real_signal_mask(size : int, bandwith : int) -> tf.Tensor:
+    """
+    Create a low pass filter mask to be applied on a real signal.
+    Since the DFT of a real signal is Hermitian-symmetric, only returns the fft_length / 2 + 1
+    unique components of the FFT.
+
+    Parameters
+    ----------
+    size
+        Size of the mask.
+    bandwith
+        Bandwith of the low pass filter. The higher the bandwith, the more frequencies are kept.
+
+    Returns
+    -------
+    mask
```

Comment:
Am I right if I say that this object has always the same number of dimension ? In that case, it could be nice to specified the output's shape.